assets/model_monitoring/components/src/action_analyzer/action_analyzer_identify_problem_traffic/run.py

```diff
@@ -187,13 +187,16 @@ def parse_debugging_info(root_span):
                 for document in retrieval_documents:
                     text.append(document["document.content"])
                     score.append(float(document["document.score"]))
-                spans_array.append((parent_id, index_content, index_id, query, TEXT_SPLITTER.join(text), max(score)))
+                #[fep] why do we use parent_id as the span_id?
+                spans_array.append((parent_id, index_content, index_id, query, TEXT_SPLITTER.join(text), max(score)))  #[fep] is the max(score) used later on?
         return spans_array
     except KeyError as e:
         print("Required field not found: ", e)
         return None
 
-
+#[fep] it will be hard to extend this function in the future. because the output schema is tie to index span too much
```

Comment:
can check offline

Reply:
I remember in the preprocessor, they have a logic to expand the Input object to multiple fields without knowing the detailed schema inside the Input. Maybe this logic can help?